crates/apollo-router/src/lib.rs

```diff
@@ -424,6 +424,10 @@ pub struct FederatedServer {
     /// The Configuration that the server will use. This can be static or a stream for hot reloading.
     configuration: ConfigurationKind,
 
+    /// Limit query cache entries
+    #[builder(default = 100)]
+    query_cache_limit: usize,
```

Comment:
1.  This should go in `Configuration`

2.  I'm not convinced this is the right name for this field...
    
    This variable in particular concerns the cache size of the query planning. There will probably be more caches. For example the queries will be parsed to `Query` objects Soon :tm: and it's also kind of a query cache.

    I think we need to be more specific. For example we could create a struct `Cache` (in module `configuration`) that will have a field for each different cache. What do you think?

Reply:
I know what you mean about the name and the fact that it should be in configuration (which didn't exist before my merge).
Name suggestion: plan_cache_limit (to identify that it is only for plans). I could make that change in this PR.

The bigger suggestion of changing to Configuration and the Cache struct should be in a separate Issue/PR I think, since it is probably expanding the scope of this one too much.

Reply:
Configuration exists for a very long time now :thinking: even before I arrived.

> Name suggestion: plan_cache_limit (to identify that it is only for plans).

Sounds good to me

> The bigger suggestion of changing to Configuration and the Cache struct should be in a separate Issue/PR I think, since it is probably expanding the scope of this one too much.

I don't think we are in a hurry and since you are introducing those I would prefer that it is included in this PR. Things that are out-of-scope imo should be things that exist already in the code and should be improved in some way. But the parameter plan_cache_limit is something introduced in this PR so it makes sense to me that this PR locates it as best it could. I'm pretty sure this will also simplify your code.

It is possible there is a reason I'm not seeing here that makes this change complicated somehow so feel free to merge if that's the case.

Reply:
Sorry, that was my mis-apprehension. I was trying to refer to the changes you had made to internal parameters yesterday.

I could add it as a configuration, but I slightly prefer a command line and I think it could be resolved in a later issue/PR.

I think there's a good discussion here which will keep to a later issue/PR, so I will merge this and preserve the discussion in issue #188

Reply:
Resolving as moved to #188